_posts/2020-07-21-cube-functionality-to-kartothek.markdown

```diff
@@ -0,0 +1,171 @@
+>>> from kartothek.core.cube.conditions import C
+>>> from kartothek.io.eager_cube import query_cube
+>>> query_cube(
+...     cube=cube,
+...     store=store,
+...     payload_columns=["avg_temp"],
+...     conditions=(
+...         (C("country") == "DE") &
+...         C("latitude").in_interval(50., 52.) &
+...         C("longitude").in_interval(13., 14.)
+...     ),
+... )[0]
+   avg_temp     city country        day
+0         8  Dresden      DE 2020-07-01
+1         4  Dresden      DE 2020-07-02
+```
+
+
+## Additional Features of Cube
+
+*	**Multi-dataset with Single Table**: When mapping multiple parts (tables or datasets) to Kartothek, using multiple datasets allows users to copy, backup and delete them separately. 
+Index structures are bound to datasets which feels more consistent.
+*	**Explicit physical Partitions**: We have decided for explicit physical partitions since we have seen that this data model works well for our current data flow. 
+It allows quick and efficient re-partitioning to allow row-based, group-based, and timeseries-based data processing steps, while keeping the technical complexity rather low (compared to an automatic + dynamic partitioning).
+It also maps well to multiple backends we planned to use.
+*	**Update Granularity Partition-wise**: Entire partitions can overwrite old physical partitions. Deletion operations are partition-wise.
+*	**Seed-Based Join System / Partition-alignment**: When data is stored in multiple parts (tables or datasets), the question is how to expose it to the user during read operations.
+ Seed based Join marks a single part as seed which provides seed dataset in the cube, all other parts are just additional columns.
+ Cube uses lazy approach of seed based join, 
+ since it better supports independent copies and backups of datasets and also simplifies some of our processing pipelines (e.g. geolocation data can blindly be fetched for too many locations and dates.)	
+
+## Dask And Yamal Support
+
+The following features are supported by different backends:
+
+
+| Feature | Eager | Dask.Bag | Dask.DataFrame |	Yamal |
+| :-------: | :-----: | :--------: | :--------------: | :-----: |
+| Build	  |    yes    |    yes     |      yes         |   yes   |
+| Extend  |	 yes  |	  yes	 |     yes        |	 yes  |
+| Remove  |  yes  |   no	 |     no         |   no  |
+| Append  |  yes  |   yes	 |     yes        |	 yes  |
+| Delete  |  yes  |   yes    |     no         |  yes  |
+| Copy	  |  yes  |   yes    |	   no         |	 yes  |
+| Query	  |  yes  |   yes    |	   yes        |  yes  |
+| Stats	  |  yes  |   yes	 |     no	      |  yes  |
+| Cleanup |	 yes  |   yes    |	   no         |  yes  |
+| ------- | ----- | -------- | -------------- | ----- |
+
+
+## New Features to Kartothek :
+Kartothek features a **command line interface (CLI)** for some cube operations. 
+To use it, create a  **skv.yml** file that describes storefact stores and use commands to gather information of the required cube.
```

Comment:
What should be part of the `skv.yml`? Maybe cross-reference [storefact](https://github.com/JDASoftwareGroup/storefact/) since this is where the store descriptions come from.

Reply:
done.